packages/asl-projects/client/helpers/trainingRecordsComparison.js

```diff
@@ -0,0 +1,222 @@
+    const diffGray = detectChanges(cur, first);
+
+    if (diffPink) {
+      changedPink.push(cur.trainingId);
+      changedPinkDetails[cur.trainingId] = diffPink;
+    }
+    if (diffGray) {
+      changedGray.push(cur.trainingId);
+      changedGrayDetails[cur.trainingId] = diffGray;
+    }
+  });
+
+  results.changed.push({ color: 'pink', ids: changedPink, details: changedPinkDetails });
+  results.changed.push({ color: 'gray', ids: changedGray, details: changedGrayDetails });
+
+  return results;
+}
+
+/**
+ * Safely find the highlight colour for a record ID within a group of change arrays.
+ *
+ * @param {Array<Object>} groups - Array of change groups (added/removed/changed).
+ * @param {string|number} id - Training record ID.
+ * @returns {string|null} - The matching highlight colour (e.g., "pink" or "gray").
+ */
+export const findColor = (groups = [], id) => {
+  if (!id || !Array.isArray(groups)) return null;
+  const idStr = String(id).trim();
+
+  const group = groups.find(g =>
+    Array.isArray(g?.ids) && g.ids.some(x => String(x).trim() === idStr)
+  );
+
+  return group?.color || null;
+};
+
+/**
+ * Returns status metadata for rendering training record change badges.
+ *
+ * @param {Object} record - The training record object (with `trainingId` or `id`).
+ * @param {Object} comparisons - The comparison result object (from compareTrainingRecords()).
+ * @returns {Object|null} - Badge data { label, class } or null if unchanged.
+ */
+export const getStatus = (record, comparisons = {}) => {
+  if (!record || typeof record !== 'object') return null;
+
+  const { added = [], removed = [], changed = [] } = comparisons;
+  const recordId = String(record.trainingId || record.id || '').trim();
+
+  const addedColor = findColor(added, recordId);
+  const removedColor = findColor(removed, recordId);
+  const changedColor = findColor(changed, recordId);
+
+  // Priority: changed > removed > added
+  if (changedColor) {
+    return { label: 'CHANGED', class: `badge changed ${changedColor}` };
+  }
+  if (removedColor) {
+    return { label: 'REMOVED', class: `badge deleted ${removedColor}` };
+  }
+  if (addedColor) {
+    return { label: 'NEW', class: `badge created ${addedColor}` };
+  }
+
+  return null;
+};
+
+/**
+ * Retrieve a specific training record from versioned history.
+ *
+ * @param {Array<Object>} data - Array of version objects (each containing trainingRecords[]).
+ * @param {string} trainingId - ID of the record to locate.
+ * @param {'current'|'previous'|'first'} versionType - Which version to fetch from.
+ * @returns {Object|null} - The training record object, or null if not found.
+ */
+export const getTrainingRecord = (data = [], trainingId, versionType = 'current') => {
+  if (!Array.isArray(data) || !trainingId) return null;
+
+  const versionsWithRecord = data.filter(
+    v => Array.isArray(v.trainingRecords) &&
+      v.trainingRecords.some(r => r.trainingId === trainingId)
+  );
```

Comment:
If a version doesn't have a record it should still be included and the training record assumed to be an empty array